### DISCUSSION TITLE
g.proj symbol lookup error

### DISCUSSION THREAD

**Discussion by Juuuleuh-le-canare** [OSGeo/grass#2954] [Q&A]
Hello everyone,

**--Problem Description**
I am trying to use grass within a python code thanks to the grass_session and grass_scirpt packages.

I am not very familiar with the "session" part. From what i understand i need to specify where and how will be named the file architecture my grass functions will work in. I am not sure which one fo GISDBASE/LOCATION... is for me to choose, and which one must point to a specific directory.

As for now my gisbase variable is pointing to where grass 7.8.5 is installed on my computer, that is to say: "usr/lib/grass78"
And i set arbitrary names for "location" and "mapset"

**--The Error**
```
RuntimeError: Cannot create: /home/mambauser/data_agata/grassdata/abancay_peru with the following options: EPSG:32718. Executing:
grass -c EPSG:32718 -e /home/mambauser/data_agata/grassdata/abancay_peru
GRASS said:
b''
b"System locale is not usable (LC_ALL variable not defined). Most likely it indicates misconfigured environment.\nReported error message: unsupported locale setting\nDefault locale settings are missing. GRASS running with C locale.\nStarting GRASS GIS...\nCreating new GRASS GIS location <abancay_peru>...\nERROR: b'g.proj: symbol lookup error: /usr/lib/libgdal.so.28: undefined symbol: opj_encoder_set_extra_options\n\nExiting...\n"

```

**--Code**
```
class texture:
  
  def __init__(self,filename):
    year=filename[-4:]
    self.path_corona="/home/mambauser/data_agata/"+str(year)+"/Images/"+str(filename)+"/"
    self.path_mask="/home/mambauser/data_agata/"+str(year)+"/Images/"+str(filename)+"/"  +str(filename)
    self.path_mnt='//home/mambauser/data_agata/MNT'
    self.path_indices = "/home/mambauser/data_agata/"+str(year)+"/Indices/"
    

    self.raster_fname=str(filename)
    self.corrected_raster=self.path_corona+str(filename)+'.tif'
    self.used_cores=int(multiprocessing.cpu_count()*2/3)



def haralick_grass(self,filename,bande,nom_bande,fenetre):
    entree= self.corrected_raster
      
    #sortie A changer pour aboutir dans dossier INDICES + ajouter taille radius
    sortie=self.path_indices + str(filename)+'_haralick_{}.tif'.format(fenetre)
    
    #Commebnt gère multibande?
    # settings
    gisbase="/usr/lib/grass78"
    gisdb="/home/mambauser/data_agata/grassdata"
    location="abancay_peru"
    mapset="user_jules"

    rcfile = gsetup.init(gisbase, gisdb, location, mapset)

    PERMANENT = Session()
     
    PERMANENT.open(gisdb=gisdb, location=location,
                create_opts="EPSG:32718")
     
    PERMANENT.close()
    
    with Session(gisdb=gisdb, location=location, mapset=mapset,
                create_opts="EPSG:32718"):
                 
      #r.texture assumes grey levels ranging from 0 to 255 as inpu
       grass.run_command('r.texture',
               flags="averbose",
               input = entree,
               output = sortie,
             size=fenetre,
              )
    return

fname="corona_30061966"

calcul=texture(fname)
calcul.compute_index(fname)
```

**Comment by marisn** [comment]
Your problem is broken GDAL library. Check if you don't have two of them (and GRASS is picking up the wrong one).

**Reply by Juuuleuh-le-canare** [reply]
I do have two different version of gdal on my computer. The first one  is complete, and the second one is incomplete and the one grass is trying to use based my environment variables, as far as i understand it.

The smaller (second) gdal version is necessary for a library called otb that I use. Iwould like to avoid "breaking it" by changing the names of the environement variables (attached file) in order for grass to work.

What would you suggest?

[otbdocker_env_var.txt](https://github.com/OSGeo/grass/files/11449747/otbdocker_env_var.txt)
![second gdal](https://github.com/OSGeo/grass/assets/101324701/cc92b64b-7871-44eb-9acf-91d78d555d8a)
![first gdal](https://github.com/OSGeo/grass/assets/101324701/43e4f947-e1d2-4248-81fa-d17aa598de23)

**Reply by marisn** [reply]
It is possible to run with multiple GDAL versions. It is necessary just to ensure that each version gets only its own dependencies. Examine your LD_LIBRARY_PATH variable and also take look at the output of ldd for libgrass_raster.so and libgdal.so. Some of libs loaded will be either missing or pointing to wrong files. Then you'll just have to adjust your environment till you get the correct ones listed.

**Comment by neteler** [comment]
> As for now my gisbase variable is pointing to where grass 7.8.5 is installed on my computer, that is to say: "usr/lib/grass78"

If all possible, please consider to upgrade to GRASS GIS 8. It comes with a WAY better Python support.

**Reply by Juuuleuh-le-canare** [reply]
I tried to install the most recent version avaible withe the following command lines:
`sudo add-apt-repository ppa:ubuntugis/ppa
sudo add-apt-repository ppa:grass/grass-stable
sudo apt-get update`

And i end up with grass 7.8.5, as it seems to be the stable version for my distribution that is 
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye

I am working on a remote docker so i cannot update/change my distribution.
